File: src/main/java/com/chukapoka/server/treemember/controller/MemoController.java
package com.chukapoka.server.treemember.controller;

import com.chukapoka.server.treemember.domain.Memo;
import com.chukapoka.server.treemember.domain.ResponseData;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.ArrayList;
import java.util.List;

@RestController
@RequestMapping("/api/memo")
public class MemoController {

    private final ArrayList<Memo> list = new ArrayList<Memo>() {
        {
            add(new Memo(0, "content0", false));
            add(new Memo(1, "content1", false));
        }
    };

    @GetMapping("")
    public ResponseEntity<ArrayList<Memo>> list() {
        return ResponseEntity.ok(list);
    }

    @PostMapping("")
    public ResponseEntity<ResponseData> newItem() {
        int id = list.size();
        Memo newMemo = new Memo(id, "", false);
        list.add(newMemo);
        return ResponseEntity.ok(new ResponseData(id, true));
    }

    @PutMapping("/{id}")
    @CrossOrigin
    public ResponseEntity<ResponseData> put(@PathVariable int id, @RequestBody Memo memo) {
        List<Memo> newList = list.stream().map(item -> item.id == id ? memo : item).toList();
        list.clear();
        list.addAll(newList);
        return ResponseEntity.ok(new ResponseData(id, true));
    }
}

File: src/main/kotlin/com/chukapoka/server/controller/MemoController.kt
package com.chukapoka.server.controller

import com.chukapoka.server.domain.Memo
import com.chukapoka.server.domain.ResponseData
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/memo")
class MemoController {

    private val list =  mutableListOf<Memo>().apply {
        repeat(2){ this.add( Memo(
                id = it,
                content = "content$it",
                isDone = false,
        )) }
    }

    @GetMapping("")
    fun list() : ResponseEntity<List<Memo>> {
        return ResponseEntity.ok(list.toList())
    }

    @PostMapping("")
    fun new() : ResponseEntity<ResponseData> {
        val id = list.size
        val newMemo = Memo(id=id, content = "", isDone = false)
        list.add(newMemo)
        return ResponseEntity.ok(ResponseData(id, true))
    }

    @PutMapping("/{id}")
    @CrossOrigin
    fun put(@PathVariable id: Int , @RequestBody memo: Memo) : ResponseEntity<ResponseData> {
        val newList = list.map { if (it.id == id) memo else it }
        list.clear()
        list.addAll(newList)
        return ResponseEntity.ok(ResponseData(id, true))
    }
}
